src/agent.cpp

```diff
@@ -1036,28 +1073,42 @@ bool Agent::doReduceAngleToGoal() {
   double dist_to_end2 = targetLineEnd2.dist2(ball_pos);
   double ratio = dist_to_end2/(dist_to_end1+dist_to_end2);
   Vector2D target = targetLineEnd1 * ratio + targetLineEnd2 * (1-ratio);
-  Body_GoToPoint(target, 0.25, ServerParam::i().maxDashPower()).execute(this);
-  return true;
+  bool may_fix = wm.self().collidesWithPost();
+  if (Body_GoToPoint(target, 0.25, ServerParam::i().maxDashPower()).execute(this)) {
```

Comment:
Can we change this to:
if (Body_GoToPoint(target, 0.25, ServerParam::i().maxDashPower()).execute(this) || wm.self().collidesWithPost()) {
  return true;
}
return false;